File: front/src/components/customer/components/commons-components/dtos/customer-dto.ts

import { RootBase } from 'shared/entities-dtos/root-base';
import { EntityTypeEnumDto } from '../../commons-components/dtos/entity-type.enum-dto';
import { AddressDto } from 'shared/components/address/dtos/address-dto';
import { ContactDto } from 'shared/components/contact/dtos/contact-dto';


export class CustomerDto extends RootBase {

  constructor() {super()}

  name!: string;
  tradeName!: string;
  cnpj!: string;
  entityType!: EntityTypeEnumDto
  description!: string;
  address!: AddressDto;
  contact!: ContactDto;

}

File: back/Application/Services/Operations/Main/Customers/Dtos/CustomerDto.cs
using System;
using Application.Services.Operations.Main.Inheritances;
using Application.Services.Operations.Main.Inheritances.Enums;
using Application.Services.Shared.Dtos;
using Domain.Entities.Main.Customers;

namespace Application.Services.Operations.Main.Customers.Dtos
{
    public class CustomerDto : RootBaseDto
    {
        public string Name { get; set; }
        public string TradeName { get; set; }
        public string CNPJ { get; set; }
        public EntityTypeEnumDto EntityType { get; set; }
        public string Description { get; set; }
        public AddressDto Address { get; set; }
        public ContactDto Contact { get; set; }
    }
}